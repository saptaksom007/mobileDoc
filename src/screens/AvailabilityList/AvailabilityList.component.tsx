import React, { PureComponent } from 'react'
import { compose } from 'recompose'
import { FlatList } from 'react-native'
import i18n from 'ex-react-native-i18n'

import EmptyList from 'components/EmptyList/EmptyList.component'
import { connect } from 'react-redux'
import Conversation from 'components/Conversation/Conversation.component'
import { Actions as NavAction } from 'navigation/SagaNavigation'
import { getClinicsFromAvailabilities } from 'common-docdok/lib/domain/healthrelation/selectors/findClinic'
import withSubtitle from 'screens/enhancers/withSubtitle'
import { isInteger } from 'utilities/number'

import styles from './AvailabilityList.styles'
import { Action } from 'redux'

const getClinicMessage = (item: {
  doctorMessage: string
  clinicMessage: string | number
}) => {
  if (!item.clinicMessage) {
    return item.doctorMessage
  }
  return isInteger(item.clinicMessage)
    ? i18n.t('availabilitydetail.card.clinicMessage', {
      hours: item.clinicMessage
    })
    : (item.clinicMessage as string)
}

interface Props {
  dispatch(action: Action): void
  clinics: any[]
}

export class AvailabilityList extends PureComponent<Props> {
  renderRow = ({ item }: any) => (
    <Conversation
      title={item.clinicName}
      onPress={() =>
        this.props.dispatch(NavAction.push('availabilitydetail', { ...item }))
      }
      content={getClinicMessage(item)}
      hideAvatar
      hideChevron={false}
      detailNumberOfLines={4}
      iconName={'access-time'}
    />
  )

  render() {
    const { clinics } = this.props

    if (!clinics || clinics.length === 0) {
      return <EmptyList show />
    }

    return (
      <FlatList
        contentContainerStyle={styles.container}
        data={clinics}
        renderItem={this.renderRow}
        keyExtractor={(_item, index) => `avail-${index}`}
      />
    )
  }
}

function select({ healthrelation: { availabilities } }: any) {
  const clinics = getClinicsFromAvailabilities(availabilities) || []
  return { clinics }
}

export default compose<Props, any>(
  connect(select),
  withSubtitle('availabilityList.subtitle')
)(AvailabilityList)
