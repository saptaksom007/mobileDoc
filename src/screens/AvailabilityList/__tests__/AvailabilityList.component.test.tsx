import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'
import { AvailabilityList } from '../AvailabilityList.component'

it('<AvailabilityList/> render correctly', () => {
  const tree = renderer
    .create(<AvailabilityList clinics={[]} dispatch={() => {}} />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
