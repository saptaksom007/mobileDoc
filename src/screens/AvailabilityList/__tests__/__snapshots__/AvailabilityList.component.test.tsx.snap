// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AvailabilityList/> render correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "height": 1294,
      "justifyContent": "center",
    }
  }
>
  <Text
    allowFontScaling={false}
    ellipsizeMode="tail"
    onPress={[Function]}
    style={
      Array [
        Object {
          "fontFamily": "TitilliumWeb-Regular",
          "fontSize": 17.5,
          "lineHeight": 22,
          "overflow": "visible",
        },
        Object {
          "color": "rgba(27, 106, 158, 0.85)",
          "fontSize": 17.5,
          "lineHeight": 21,
        },
        Object {
          "color": "#999",
          "marginTop": -40,
        },
      ]
    }
  >
    global.empty
  </Text>
</View>
`;
